packages/core/src/binary-data/object-store/object-store.service.ee.ts

```diff
@@ -243,28 +220,37 @@ export class ObjectStoreService {
 
 	private async request<T>(
 		method: Method,
-		host: string,
 		rawPath = '',
 		{ qs, headers, body, responseType }: RequestOptions = {},
 	) {
-		const path = this.toPath(rawPath, qs);
+		const url = new URL(this.baseUrl);
+		if (rawPath && rawPath !== '/') {
+			url.pathname = `${url.pathname}/${rawPath}`;
+		}
+		Object.entries(qs ?? {}).forEach(([key, value]) => {
+			url.searchParams.set(key, String(value));
+		});
 
 		const optionsToSign: Aws4Options = {
 			method,
 			service: 's3',
-			region: this.bucket.region,
-			host,
-			path,
+			region: this.s3Config.bucket.region,
+			host: this.s3Config.host,
+			path: url.pathname,
```

Comment:
> [pathname](https://developer.mozilla.org/en-US/docs/Web/API/URL/pathname)
>
>    A string containing an initial '/' followed by the path of the URL, not including the query string or fragment.

This now excludes the query params. I think they were included before? Should we use .toString() here instead?

Reply:
good catch. I'll update the tests to assert on complete headers, make it pass on `master`, and then make this branch pass the same tests.